DietPi-Software | Fix ipfs command when called from non-root user

nonnorm opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready

**Commit list/description**:
<!--
- DietPi-Config | Add "Fan control" option to "Performance Options"
![Screenshot](https://xxx.github.com/images/xxx.png)
- DietPi-Config | Add "Fan control" support for Odroid C2
- DietPi-Config | Syntax fix
-->
- DietPi-Software | Allow any user with sudo permissions to use the ipfs command
- DietPi-Patch_file | Allow dietpi user to run sudo commands as any user

nonnorm committed:
v7.3
+ DietPi-Software | Allow any user with sudo permissions to use the ipfs command

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -9419,7 +9419,7 @@ _EOF_
 
 			# CLI alias
-			echo "alias ipfs='sudo -u ipfs IPFS_PATH=/mnt/dietpi_userdata/ipfs ipfs'" > /etc/bashrc.d/dietpi-ipfs.sh
+			echo "alias ipfs='sudo sudo -u ipfs IPFS_PATH=/mnt/dietpi_userdata/ipfs ipfs'" > /etc/bashrc.d/dietpi-ipfs.sh
```
Now there is two times `sudo`, which is not wanted I think? 😄

Ah or you mean that some users may be allowed to call `sudo (-u root)` but not with another target user than root? But that should be better fixed via `sudoers` or by simply running `usermod -aG sudo <username>` instead of cascading `sudo` calls by default. That spams the logs and creates nested subshells as well, being slow etc.

nonnorm replied:
Is that better?  I'm honestly not sure which patch file to use.

nonnorm committed:
v7.3
+ DietPi-Patch_file | Allow dietpi user to run sudo commands as any user

MichaIng commented on dietpi/patch_file:
```diff
@@ -16,6 +16,9 @@
 	#////////////////////////////////////
 
+	# Allow dietpi user to run sudo commands as any user
+	echo 'dietpi ALL=(ALL) NOPASSWD: ALL' > /etc/sudoers.d/dietpi
```
Currently: https://github.com/MichaIng/DietPi/blob/master/dietpi/func/dietpi-set_software#L407
- `dietpi ALL=NOPASSWD: ALL`

This does not allow becoming any user? I'll test later. Else we should change that in the underlying script as well.

nonnorm replied:
No, it doesn't.  From the `sudoers` man page:
```
If no Runas_Spec is specified the command may be run as root and no group may be specified.
```

MichaIng replied:
Ah okay. Then you last commit makes sense. As long as `sudo sudo -u ...` can be used to become any user, this does not really add any permissions, it just allows to become any user without this awkward `sudo` concatenation 😄.

nonnorm committed:
v7.3
+ DietPi-Set_Software | Let dietpi user become any user with sudo

MichaIng committed:
v7.3

+ DietPi-Patch | Do not use old patch file

MichaIng committed:
v7.3

+ DietPi-Patches | Update dietpi sudoers file to allow sudo calls with non-root users without manual password input

MichaIng committed:
v7.3

+ DietPi-Patches | Allow "-g" as well and fix comment

MichaIng committed:
v7.3

+ DietPi-Set_software | useradd: Allow "sudo -g <group>" as well
+ DietPi-Set_software | userdel: Remove obsolete pre-v6.29 code as we completely overwrite the file on v7.3 patches
+ DietPi-Set_software | Add G_EXEC in more cases where applicable
